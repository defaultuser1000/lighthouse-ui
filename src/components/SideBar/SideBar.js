import React from 'react';
import SideBarItem from './SideBarItem/SideBarItem';
import {Divider, Menu} from 'semantic-ui-react';
import './SideBar.scss';
import {SideBarHeader} from "./SideBarHeader/SideBarHeader";
import {SideBarFooter} from "./SideBarFooter/SideBarFooter";

export default class SideBar extends React.Component {
    render() {
        return (
            <Menu borderless vertical stackable fixed='left' className='side-nav'>
                <SideBarItem path='/' label='Home' icon='home'/>
                <SideBarItem path='/orders' label='Orders' icon='film'/>
                <SideBarItem path='/users' label='Users' icon='user outline'/>
                <SideBarItem path='/schedule' label='Working Schedule' icon='calendar outline'/>
                <Divider/>
                <SideBarHeader title='System'/>
                <SideBarItem path='/settings' label='Settings' icon='settings'/>

                <SideBarFooter style={{verticalAlign: 'baseline'}}/>
            </Menu>
        );
    }
}