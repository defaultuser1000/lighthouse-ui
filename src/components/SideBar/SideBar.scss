@import '../../styles/shared.scss';
$sidebar-hover-color: #ebebeb;
.ui.menu.fixed.side-nav {
  background-color: #f0f0f0;
  margin-top: $header-nav-height;
  padding-bottom: $header-nav-height;
  overflow-y: auto;

  .sidebar-item:hover {
    background: $sidebar-hover-color;
  }
}

.side-nav.ui.vertical.menu {
  width: $sidebar-left-width;
}