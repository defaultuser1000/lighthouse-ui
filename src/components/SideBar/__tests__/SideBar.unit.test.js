import React from 'react';
import { shallow } from 'enzyme';
import SideBar from '../SideBar';

test('renders HeaderNav', () => {
    const wrapper = shallow(
        <SideBar />
    );
    expect(wrapper).toMatchSnapshot();
});