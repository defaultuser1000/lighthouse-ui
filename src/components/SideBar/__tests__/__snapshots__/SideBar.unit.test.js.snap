// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeaderNav 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <SideBarItem
    highlight={true}
    icon="home"
    label="Home"
  />
  <SideBarItem
    icon="film"
    label="Order"
  />
  <SideBarItem
    icon="user outline"
    label="Users"
  />
  <SideBarItem
    icon="calendar outline"
    label="Working Schedule"
  />
  <Divider />
  <SideBarHeader
    title="System"
  />
  <SideBarItem
    icon="settings"
    label="Settings"
  />
  <SideBarFooter />
</Menu>
`;
